snappy-core/src/main/scala/org/apache/spark/sql/streaming/FileStreamSource.scala

```diff
@@ -62,6 +61,9 @@ final class FileStreamRelation(
   val directory = options(DIRECTORY)
 
   // TODO: Yogesh, add support for other types of files streams
-  override protected def createRowStream(): DStream[InternalRow] =
+  override protected def createRowStream(): DStream[InternalRow] = {
+    val converter = CatalystTypeConverters.createToCatalystConverter(schema)
     context.textFileStream(directory).flatMap(rowConverter.toRows)
+        .map(converter(_).asInstanceOf[InternalRow])
+  }
 }
```

Comment:
@ymahajan perhaps we can have something similar BaseRelation.needConversion for some possible optimization in this. If the implementation says needConversion is false, then assume the converter returns InternalRows and treat it likewise (i.e. returning a Seq[InternalRow] even though the signature says Seq[Row]).